scmail.py

```diff
@@ -0,0 +1,205 @@
+def load_config(ctx):
+    """Load config file."""
+    print("Now, laod previous config file:")
+    f = ctx.obj['config'].open('r').read()
+    config = re.split('\n|=', f)
+
+    for i in range(0, len(config), 2):
+        print('The {} is: {}'.format(config[i], config[i + 1]))
+
+    print('\n')
+    return config[1], config[3]
+
+
+@click.pass_context
+def set_user_info(ctx, gnupghome, email):
+    """Set user information and return gnupghome and email."""
+    config = 'gnupghome=' + gnupghome + '\n' + 'email=' + email
+
+    # Store config file and other information.
+    ctx.obj['config'].touch()
+    with ctx.obj['config'].open('w') as f:
+        f.write(config)
+
+    return gnupghome, email
+
+
+@click.command()
+@click.option('--gnupghome', prompt='Enter your gnupg home dir', default='gnupgkeys',
+              help='The home dir of gnupg key pairs.')
+@click.option('--email', prompt='Enter your email', default=getlogin() + '@scmail.dev', help='The email of user.')
+@click.pass_context
+def update_user_info(ctx, gnupghome, email):
+    """Update gnupg home dir and personal email"""
+    pass
+
+
+@click.command()
+@click.option('--key-type', prompt='Enter key type', default="RSA", help='The algorithms to generate the key. ex. RSA')
+@click.option('--key-length', prompt='Enter key length', default=1024, help='The length of the key.')
+@click.option('--expire-date', prompt='Enter expire date', default="2y", help='The expire time of the key pair.')
+@click.password_option('--password', prompt='Enter private key password', help='Password of private key.')
+@click.pass_context
+def create_key(ctx, key_type, key_length, expire_date, password):
+    """Create gnupg key pairs."""
+    print('Those are the information about the key will be created:\n')
+    ctx.obj['pp'].pprint({'email': ctx.obj['gpg'].email, 'key type': key_type, 'key length': key_length, 'expire date': expire_date})
+    print('\n')
+    key = ctx.obj['gpg'].create(password, key_type=key_type, key_length=key_length, expire_date=expire_date)
+    ctx.obj['pp'].pprint(key.__dict__)
+
+
+@click.command()
+@click.option('--show-private', prompt='Do you want to show the private keys?', default=False, is_flag=True,
+              help='Whether show the private keys')
+@click.pass_context
+def list_keys(ctx, show_private):
+    """List and print all the keys in the gnupg home dir."""
+    keys = ctx.obj['gpg'].list_keys(show_private)
+    key_type = 'public' if show_private is False else 'private'
+    print('You have {} {} keys.'.format(len(keys), key_type))
+    ctx.obj['pp'].pprint(keys.__dict__)
+
+
+@click.command()
+@click.pass_context
+def register(ctx):
+    """Register sc_mail API."""
+    # Request register.
+    fingerprint = ctx.obj['gpg'].list_keys(True).curkey.get('fingerprint')
+    data = {'fingerprint': fingerprint}
+
+    # Register
+    address = SECUREMAILBOX_URL + '/register/'
+    r = requests.post(address)
+    print(r.text)
+
+    if 'success' in r:
```

Comment:
`success` might always be in `r` regardless of whether it's `True` or not. This should probably check that `r.status_code == 200` in addition to also ensuring that `success` is `True`.

Reply:
Thanks, I looked up the code of API, and look like the `error` field, not `success` will be returned when an error occurred. However, `status_code` should be added to identify return status.